File: Task2/apz-pzpi-21-11-milennyi-ivan-task2/src/models/feed_supply.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use utoipa::ToSchema;
use validator::Validate;

#[derive(Debug, Validate, Deserialize, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedSupply{
    id: Option<u64>,
    storekeeper_id: Option<u64>,
    #[validate(range(min = 1))]
    amount: u64,
    timestamp: u64,
    feed_id: u64
}

impl FeedSupply {
    pub fn id(&self) -> Option<u64> {
        self.id
    }

    pub fn storekeeper_id(&self) -> Option<u64> {
        self.storekeeper_id
    }

    pub fn amount(&self) -> u64 {
        self.amount
    }
    pub fn timestamp(&self) -> u64 {
        self.timestamp
    }
    pub fn feed_id(&self) -> u64 {
        self.feed_id
    }
}

File: Task4/apz-pzpi-21-11-milennyi-ivan-task4/src/models/FeedSupply.ts
export class FeedSupply {
    id: number | null;
    storekeeper_id: number | null;
    amount: number;
    timestamp: number;
    feed_id: number;

    constructor(id: number | null, storekeeperId: number | null, amount: number, timestamp: number, feedId: number) {
        this.id = id;
        this.storekeeper_id = storekeeperId;
        this.amount = amount;
        this.timestamp = Math.floor(timestamp / 1000);
        this.feed_id = feedId;
    }
}